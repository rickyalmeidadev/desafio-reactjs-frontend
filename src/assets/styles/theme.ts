export default {
  colors: {
    light: '#fff',
    dark: '#212121',
    gray: '#9E9E9E',
    shade: '#424242',
  },
  margins: {
    smaller: '4px',
    small: '8px',
    normal: '16px',
    large: '24px',
    larger: '32px',
  },
  paddings: {
    smaller: '4px',
    small: '8px',
    normal: '14px',
    large: '24px',
    larger: '32px',
  },
  maxWidth: '1132px',
  transition: 'all 200ms ease',
};
