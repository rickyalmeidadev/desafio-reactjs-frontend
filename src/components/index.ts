export { default as Field } from './Field';
export { default as Button } from './Button';
export { default as NaverForm } from './NaverForm';
export { default as NaverModal } from './NaverModal';
export { default as DeleteModal } from './DeleteModal';
export { default as SuccessModal } from './SuccessModal';
export { default as Header } from './Header';
export { default as NaverCard } from './NaverCard';
export { default as ModalWrapper } from './ModalWrapper';
export { default as NaverCardSkeleton } from './NaverCardSkeleton';
export { default as NaverModalSkeleton } from './NaverModalSkeleton';
