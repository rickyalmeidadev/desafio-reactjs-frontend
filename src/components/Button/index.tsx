import React, { ButtonHTMLAttributes } from 'react';

import { CustomButton } from './styles';

interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
  fullWidth?: boolean;
  noFill?: boolean;
}

const Button: React.FC<Props> = ({ children, fullWidth, noFill, ...rest }) => {
  return (
    <CustomButton {...rest} fullWidth={fullWidth} noFill={noFill}>
      {children}
    </CustomButton>
  );
};

export default Button;
