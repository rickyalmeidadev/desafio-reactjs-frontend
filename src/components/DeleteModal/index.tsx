import React from 'react';
import { useModal } from '../../hooks/useModal';

import { Container } from './styles';
import { ModalWrapper, Button } from '..';

const DeleteModal: React.FC = () => {
  const { deleteToggle, handleDeleteToggle } = useModal();

  return (
    <ModalWrapper isShowing={deleteToggle}>
      <Container>
        <h2>Excluir Naver</h2>
        <p>Tem certeza que deseja excluir este Naver?</p>

        <section>
          <Button noFill type="button" onClick={handleDeleteToggle}>
            Cancelar
          </Button>
          <Button type="button">Excluir</Button>
        </section>
      </Container>
    </ModalWrapper>
  );
};

export default DeleteModal;
