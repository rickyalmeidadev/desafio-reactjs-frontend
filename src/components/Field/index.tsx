import React, { InputHTMLAttributes } from 'react';

import { FieldProps, ErrorMessage, Field as FormikField } from 'formik';
import GridLoader from 'react-spinners/GridLoader';

import { Wrapper, Label, Input, FieldError, FieldLoading } from './styles';
import { useNavers } from '../../hooks/useNavers';

interface IProps extends InputHTMLAttributes<HTMLInputElement> {
  label: string;
  name: string;
}

const Field: React.FC<IProps> = ({ label, name, ...rest }) => {
  const { isLoading } = useNavers();

  return (
    <Wrapper>
      <FormikField name={name}>
        {({ field }: FieldProps) => (
          <>
            <Label htmlFor={name}>{label}</Label>
            <Input
              {...rest}
              {...field}
              name={name}
              id={name}
              placeholder={label}
            />
            <ErrorMessage name={name} component={FieldError} />
            {isLoading && (
              <FieldLoading>
                <GridLoader size={2} />
              </FieldLoading>
            )}
          </>
        )}
      </FormikField>
    </Wrapper>
  );
};

export default React.memo(Field);
