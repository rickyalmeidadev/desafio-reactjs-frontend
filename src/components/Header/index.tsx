import React from 'react';
import { useHistory } from 'react-router-dom';

import { Container, Logo, Logout } from './styles';

import logo from '../../assets/images/logo.png';

const Header: React.FC = () => {
  const history = useHistory();

  const handleLogout = () => {
    localStorage.removeItem('nave.rs:token');
    history.push('/login');
  };

  return (
    <Container>
      <Logo src={logo} alt="nave.rs" />
      <Logout type="button" onClick={handleLogout}>
        Sair
      </Logout>
    </Container>
  );
};

export default Header;
