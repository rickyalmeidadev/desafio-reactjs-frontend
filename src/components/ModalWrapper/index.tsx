import React from 'react';

import { Container, Overlay } from './styles';

interface IProps {
  isShowing: boolean;
}

const ModalWrapper: React.FC<IProps> = ({ isShowing, children }) => {
  return (
    <>
      <Container isShowing={isShowing}>{children}</Container>
      <Overlay isShowing={isShowing} />
    </>
  );
};

export default ModalWrapper;
