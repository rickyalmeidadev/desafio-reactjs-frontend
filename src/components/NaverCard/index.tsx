import React from 'react';
import { ReactComponent as Trash } from '../../assets/icons/trash.svg';
import { ReactComponent as Pencil } from '../../assets/icons/pencil.svg';

import { Container, Picture, Name, Role, Icons } from './styles';

import picture from '../../assets/images/naver-example.png';

const NaverCard: React.FC = () => {
  return (
    <Container>
      <Picture src={picture} alt="Naver" />
      <Name>Juliano Reis</Name>
      <Role>Front-end Developer</Role>
      <Icons>
        <Trash />
        <Pencil />
      </Icons>
    </Container>
  );
};

export default NaverCard;
