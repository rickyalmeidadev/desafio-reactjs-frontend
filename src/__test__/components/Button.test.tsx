import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Button } from '../../components';
import theme from '../../assets/styles/theme';

describe('<Button />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <Button>Click Me</Button>
      </ThemeProvider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render children as its text', () => {
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <Button>Click Me</Button>
      </ThemeProvider>,
    );
    expect(getByText('Click Me')).toBeInTheDocument();
  });

  it('should have styles based in received props', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Button fullWidth noFill>
          Click Me
        </Button>
      </ThemeProvider>,
    );

    expect(getByTestId('button')).toHaveStyle(`
      width: 100%;
      background-color: transparent;
    `);
  });

  it('should have type based in received props', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <Button type="submit">Click Me</Button>
      </ThemeProvider>,
    );

    expect(getByTestId('button')).toHaveAttribute('type', 'submit');
  });
});
