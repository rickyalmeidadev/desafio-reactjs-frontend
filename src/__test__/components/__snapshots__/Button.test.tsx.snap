// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
<DocumentFragment>
  <button
    class="sc-AxheI gejEux"
    data-testid="button"
  >
    Click Me
  </button>
</DocumentFragment>
`;
